app/common/model/src/main/java/io/syndesis/common/model/DataShape.java

```diff
@@ -100,19 +100,16 @@
 
     class Builder extends ImmutableDataShape.Builder {
 
-        public static final String COMPRESSION_METADATA_KEY = "compression";
-        private static final String COMPRESSED_METADATA_KEY = "compressed";
-
         public Builder decompress() {
             DataShape currentShape = build();
-            if ("true".equals(currentShape.getMetadata().get(COMPRESSION_METADATA_KEY)) &&
-                    currentShape.getMetadata().getOrDefault(COMPRESSED_METADATA_KEY, "true").equals("true")) {
+            if ("true".equals(currentShape.getMetadata().get(DataShapeMetaData.COMPRESSION)) &&
+                    currentShape.getMetadata().getOrDefault(DataShapeMetaData.COMPRESSED, "true").equals("true")) {
```

Comment:
```suggestion
                    Boolean.parseBoolean(currentMetadata().get(DataShapeMetaData.COMPRESSED)) {
```

Reply:
Why do we have `COMPRESSED` and `COMPRESSION` metadata?

Reply:
COMPRESSION marks that the shape should be compressed. COMPRESSED marks that the shape already has been compressed in order to avoid duplicate compression.

Let me rename COMPRESSED to IS_COMPRESSED.